python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py

```diff
@@ -457,7 +457,7 @@ def _router_bgp_vlans_vlan(self: AvdStructuredConfigNetworkServices, vlan, tenan
             bgp_vlan["route_targets"]["import_export_evpn_domains"] = [{"domain": "remote", "route_target": vlan_rt}]
 
         vlan_evpn_l2_multicast_enabled = default(get(vlan, "evpn_l2_multicast.enabled"), get(tenant, "evpn_l2_multicast.enabled"))
-        if vlan_evpn_l2_multicast_enabled is True:
+        if vlan_evpn_l2_multicast_enabled is True and not vrf:
```

Comment:
I think,
if vlan_evpn_l2_multicast_enabled is True and not self.network_services_l3:

is more correct.

Reply:
Please add a comment stating the reason why vrf should not be present.